exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/metric_exporter/__init__.py

```diff
@@ -202,7 +207,85 @@ def export(
         **kwargs,
     ) -> MetricExportResult:
         # TODO(#2663): OTLPExporterMixin should pass timeout to gRPC
-        return self._export(metrics_data)
+        if not self.max_export_batch_size:
+            return self._export(data=metrics_data)
+
+        if len(metrics_data) <= self.max_export_batch_size:
+            return self._export(data=metrics_data)
+
+        for split_metrics_data in self._split_metrics_data(metrics_data):
+            export_result: MetricExportResult = self._export(
+                data=split_metrics_data
+            )
+            if export_result is MetricExportResult.FAILURE:
+                return export_result
```

Comment:
Should we try for all batches before returning?

Reply:
I'm assuming the result would be FAILURE if a single batch fails and SUCCESS if only all of them are successful?

Reply:
Right, Should it try the remaining batch or return early?

Reply:
In order to not change behavior it should attempt to try the remaining batch correct?

Reply:
With my latest changes it now tries the remaining batch

Reply:
And this configuration already exists for logs: https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/_logs/export/__init__.py#L146